src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/InterchangeHeader.java

```diff
@@ -34,26 +41,29 @@ public String getKey() {
     public String getValue() {
         String timestamp = DATE_FORMAT.format(translationTime);
         String formattedSequenceNumber = String.format("%08d", sequenceNumber);
-        return "UNOA:2"+"+"+sender+"+"+recipient+"+"+timestamp+"+"+formattedSequenceNumber;
+
+        return List.of(UNOA_PREFIX, sender, recipient, timestamp, formattedSequenceNumber, transferString).stream()
+                .filter(StringUtils::isNotBlank)
+                .collect(Collectors.joining(PLUS_SEPARATOR));
     }
 
     @Override
     protected void validateStateful() throws EdifactValidationException {
         if (sequenceNumber == null) {
             throw new EdifactValidationException(getKey() + ": Attribute sequenceNumber is required");
         }
-        if(sequenceNumber <= 0){
+        if (sequenceNumber <= 0) {
```

Comment:
Made a similar comment on NIAD-143. We should really check that this value is a whole number in the range [1, 99999999] are valid sequence numbers. Note that transaction id is "seven nines" while these sequence numbers are "eight nines"